kedro-datasets/kedro_datasets/api/api_dataset.py

```diff
@@ -127,10 +170,51 @@ def _execute_request(self) -> requests.Response:
     def _load(self) -> requests.Response:
         return self._execute_request()
 
-    def _save(self, data: None) -> NoReturn:
-        raise DataSetError(f"{self.__class__.__name__} is a read only data set type")
+    def _execute_save_with_chunks(
+        self,
+        json_data: List[Dict[str, Any]],
+    ) -> requests.Response:
+        chunk_size = self._save_args["chunk_size"]
+        n_chunks = len(json_data) // chunk_size + 1
+
+        for i in range(n_chunks):
+            send_data = json_data[i * chunk_size : (i + 1) * chunk_size]
+
+            self._save_args["json"] = send_data
+            try:
+                response = requests.request(**self._request_args)
+                response.raise_for_status()
+
+            except requests.exceptions.HTTPError as exc:
+                raise DataSetError("Failed to send data", exc) from exc
+
+            except OSError as exc:
+                raise DataSetError("Failed to connect to the remote server") from exc
+        return response
+
+    def _execute_save_request(self, json_data: Any) -> requests.Response:
+        self._save_args["json"] = json_data
+        try:
+            response = requests.request(**self._request_args)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as exc:
+            raise DataSetError("Failed to send data", exc) from exc
+
+        except OSError as exc:
+            raise DataSetError("Failed to connect to the remote server") from exc
+        return response
+
+    def _save(self, data: Any) -> requests.Response:
+        # case where we have a list of json data
+        if isinstance(data, list):
+            return self._execute_save_with_chunks(json_data=data)
+        try:
+            json_.loads(data)
+        except TypeError:
+            data = json_.dumps(data)
+
+        return self._execute_save_request(json_data=data)
```

Comment:
So data is expected to be JSON - but the checking is missing when it is a list here?

Reply:
At the moment, if given a list it will attempt to make a request with a slice of list of size chunk size. Do you think some kind of check should be done before making the request in the case of a list as a data input ?

Reply:
It was also my thought at first but not requiring a specific method for load or save could allow it to work with any kind of API that suits the user ?